### PR TITLE
[MiomioTv] Add new extractor

### PR DESCRIPTION
**Pull request by tiktok7**
added entry for high lag video site http://www.miomio.tv
(contains a range of content mostly in asian languages including instructional and promotional videos e.g. http://www.miomio.tv/teach/draw/)

### CODE REVIEW

File: docs/supportedsites.md [hunk 1]
```diff
@@ -244,6 +244,7 @@
  - **Mgoon**
  - **Minhateca**
  - **MinistryGrid**
+ - **Miomio.tv**
```
Comment (dstftw) [inline docs/supportedsites.md]: This file is autogenerated.

File: youtube_dl/extractor/miomio_tv.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(r'<meta\s+name="description"\s+content="\s*([^"]*)\s*"', webpage, 'title')
+        ref_path = self._search_regex(r'src="(/mioplayer/.*?)"', webpage, 'ref_path')
+        referer = 'http://www.miomio.tv{}'.format(ref_path)
```
Comment (dstftw) [inline youtube_dl/extractor/miomio_tv.py]: `{}` doesn't work in python 2.6.

File: youtube_dl/extractor/miomio_tv.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        xml_config = self._search_regex(r'flashvars="type=sina&amp;(.*?)&amp;cid=', webpage, 'xml config')
+        self._request_webpage("http://www.miomio.tv/mioplayer/mioplayerconfigfiles/xml.php?id={}&r=cc{}".format(id, 945), video_id)
```
Comment (jaimeMF) [inline youtube_dl/extractor/miomio_tv.py]: Do you really need to request that webpage? You don't use its contents.

If it's required, please add a comment.

File: youtube_dl/extractor/miomio_tv.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            entries.append({
+                'id': segment_id,
+                'title': segment_title,
+                'duration': segment_duration,
+                'url': segment_url
+            })
+
+        http_headers = {
+            'Referer': referer,
+            'Accept-Language': 'en,en-US;q=0.7,de;q=0.3',
```
Comment (jaimeMF) [inline youtube_dl/extractor/miomio_tv.py]: Are all headers required?

File: docs/supportedsites.md (hunk 1 above)
Comment (tiktok7) [inline docs/supportedsites.md]: got rid of the change -> checked out old version

File: youtube_dl/extractor/miomio_tv.py (hunk 2 above)
Comment (tiktok7) [inline youtube_dl/extractor/miomio_tv.py]: changed to {0} format

File: youtube_dl/extractor/miomio_tv.py (hunk 3 above)
Comment (tiktok7) [inline youtube_dl/extractor/miomio_tv.py]: The server starts dropping connections when it's not requested. I added a comment

File: youtube_dl/extractor/miomio_tv.py (hunk 4 above)
Comment (tiktok7) [inline youtube_dl/extractor/miomio_tv.py]: The language header is not. I've removed it

**Comment by dstftw** [comment]
Thanks, merged.
Do you want to be credited as one of the authors of youtube-dl? If so under what name?

**Comment by tiktok7** [comment]
That's very generous, thanks! My employer is very conservative so I'd prefer to omit my last name. 
How about: "Will W."

**Pull request merged**